src/odemis/driver/scpi.py

```diff
@@ -219,22 +219,22 @@ def _sendQuery(self, cmd, timeout=1):
         raise:
             IOError if no answer is returned in time
         """
-        cmd = cmd + "\r"
+        cmd = (cmd + "\r").encode('ascii')
         with self._ser_access:
-            logging.debug("Sending command %s", cmd.encode('string_escape'))
+            logging.debug("Sending command %s", to_str_escape(cmd))
             self._serial.write(cmd)
 
             self._serial.timeout = timeout
-            ans = ''
-            while ans[-1:] != '\r':
+            ans = b''
+            while ans[-1:] != b'\r':
                 char = self._serial.read()
                 if not char:
-                    raise IOError("Timeout after receiving %s" % ans.encode('string_escape'))
+                    raise IOError("Timeout after receiving %s" % to_str_escape(ans))
                 ans += char
 
-            logging.debug("Received answer %s", ans.encode('string_escape'))
+            logging.debug("Received answer %s", to_str_escape(ans))
 
-            return ans.rstrip()
+            return to_str_escape(ans.rstrip())
```

Comment:
No, `to_str_escape()` is only to be used for displaying string to the user. Use `decode("latin1")` here, to get a (unicode) string.